core/src/testUtils/groovy/nu/yona/server/test/AppService.groovy

```diff
@@ -0,0 +1,406 @@
+		assert user.vpnProfile.vpnPassword.length() == 32
+		assert user.vpnProfile.openVPNProfile.length() > 10
+	}
+
+	def assertUserWithoutPrivateData(def user)
+	{
+		assertPublicUserData(user)
+		assert user.nickname == null
+		assert user.devices == null
+		assert user.goals == null
+		assert user.vpnProfile == null
+	}
+
+	def assertResponseStatusCreated(def response)
+	{
+		assert response.status == 201
+	}
+
+	def assertResponseStatusSuccess(def response)
+	{
+		assert isSuccess(response)
+	}
+
+	def isSuccess(def response)
+	{
+		response.status >= 200 && response.status < 300
+	}
+
+	def getUser(Closure asserter, userURL, boolean includePrivateData, password = null)
+	{
+		def response
+		if (includePrivateData)
+		{
+			response = yonaServer.getResourceWithPassword(yonaServer.stripQueryString(userURL), password, yonaServer.getQueryParams(userURL) + ["includePrivateData": "true"])
+		}
+		else
+		{
+			response = yonaServer.getResourceWithPassword(userURL, password)
+		}
+		asserter(response)
+		return (isSuccess(response)) ? new User(response.responseData, password, includePrivateData) : null
+	}
+
+	def makeBuddies(requestingUser, respondingUser)
+	{
+		sendBuddyConnectRequest(requestingUser, respondingUser)
+		def acceptURL = fetchBuddyConnectRequestMessage(respondingUser).acceptURL
+		def acceptResponse = postMessageActionWithPassword(acceptURL, ["message" : "Yes, great idea!"], respondingUser.password)
+		assert acceptResponse.status == 200
+
+		def processURL = fetchBuddyConnectResponseMessage(requestingUser).processURL
+
+		// Have the requesting user process the buddy connect response
+		def processResponse = postMessageActionWithPassword(processURL, [ : ], requestingUser.password)
+		assert processResponse.status == 200
+	}
+
+	def sendBuddyConnectRequest(sendingUser, receivingUser)
+	{
+		// Send the buddy request
+		def response = requestBuddy(sendingUser.url, """{
+			"_embedded":{
+				"user":{
+					"firstName":"${receivingUser.firstName}",
+					"lastName":"${receivingUser.lastName}",
+					"mobileNumber":"${receivingUser.mobileNumber}",
+					"emailAddress":"not@used.nu"
+				}
+			},
+			"message":"Would you like to be my buddy?",
+			"sendingStatus":"REQUESTED",
+			"receivingStatus":"REQUESTED"
+		}""", sendingUser.password)
+		assert response.status == 201
+		response
+	}
+
+	def fetchBuddyConnectRequestMessage(User user)
+	{
+		// Have the other user fetch the buddy connect request
+		def response = getDirectMessages(user)
+		assert response.status == 200
+		assert response.responseData._embedded
+
+		def message = response.responseData._embedded?.buddyConnectRequestMessages[0]?.message ?: null
+		def acceptURL = response.responseData._embedded?.buddyConnectRequestMessages[0]?._links?.accept?.href ?: null
+		def rejectURL = response.responseData._embedded?.buddyConnectRequestMessages[0]?._links?.reject?.href ?: null
+
+		def result = [ : ]
```

Comment:
OK, but I don't like to have to create a method like this for every Message type, and have to look inside this method when the link is renamed or something like that, and have to update this method when adding a new link type.